hack/validate-generated-files

```diff
@@ -1,11 +1,25 @@
 #!/usr/bin/env bash
 
-set -eu -o pipefail
-
 case ${1:-} in
     '')
+	. $(dirname $0)/util
 	gogo_version=$(awk '$1 == "github.com/gogo/protobuf" { print $2 }' vendor.conf)
-	docker build --build-arg GOGO_VERSION=$gogo_version -f ./hack/dockerfiles/generated-files.Dockerfile --target validate --force-rm .
+	case $buildmode in
+	"buildkit")
+		buildctl build --frontend=dockerfile.v0 --local context=. --local dockerfile=. --frontend-opt build-arg:GOGO_VERSION=$gogo_version --frontend-opt filename=./hack/dockerfiles/generated-files.buildkit.Dockerfile
+		;;
+	"moby-buildkit")
+		export DOCKER_BUILDKIT=1
+		iidfile=$(mktemp -t docker-iidfile.XXXXXXXXXX)
+		docker build --iidfile $iidfile --build-arg GOGO_VERSION=$gogo_version -f ./hack/dockerfiles/generated-files.buildkit.Dockerfile --target validate --force-rm . || exit 1
+		iid=$(cat $iidfile)
+		docker rmi $iid
+		rm -f $iidfile
+		;;
+	*)
```

Comment:
docker-legacy?

Reply:
He's consistently used `*` for that case throughout. I don't have a particular opinion on spelling it out vs `*` but consistency should be retained in either case.

Reply:
I changed to `docker-legacy` in the `buildmode`. In the select, it shouldn't matter until we want to switch the defaults.